src/ring_theory/ring_hom_properties.lean

```diff
@@ -0,0 +1,123 @@
+    ∀ (f : S →+* T) (e : R ≃+* S) (hf : P f), P (f.comp e.to_ring_hom))
+
+variable {P}
+
+lemma respects_iso.cancel_left_is_iso (hP : respects_iso @P) {R S T : CommRing}
+  (f : R ⟶ S) (g : S ⟶ T)
+  [is_iso f] : P (f ≫ g) ↔ P g :=
+⟨λ H, by { convert hP.2 (f ≫ g) (as_iso f).symm.CommRing_iso_to_ring_equiv H,
+  exact (is_iso.inv_hom_id_assoc _ _).symm }, hP.2 g (as_iso f).CommRing_iso_to_ring_equiv⟩
+
+lemma respects_iso.cancel_right_is_iso (hP : respects_iso @P) {R S T : CommRing}
+  (f : R ⟶ S) (g : S ⟶ T)
+  [is_iso g] : P (f ≫ g) ↔ P f :=
+⟨λ H, by { convert hP.1 (f ≫ g) (as_iso g).symm.CommRing_iso_to_ring_equiv H,
+  change f = f ≫ g ≫ (inv g), simp }, hP.1 f (as_iso g).CommRing_iso_to_ring_equiv⟩
+
+lemma respects_iso.is_localization_away_iff (hP : ring_hom.respects_iso @P) {R S : Type*}
+  (R' S' : Type*) [comm_ring R] [comm_ring S] [comm_ring R'] [comm_ring S'] [algebra R R']
+  [algebra S S'] (f : R →+* S) (r : R) [is_localization.away r R'] [is_localization.away (f r) S'] :
+  P (localization.away_map f r) ↔ P (is_localization.away.map R' S' f r) :=
+begin
+  let e₁ : R' ≃+* localization.away r :=
+    (is_localization.alg_equiv (submonoid.powers r) _ _).to_ring_equiv,
+  let e₂ : localization.away (f r) ≃+* S' :=
+    (is_localization.alg_equiv (submonoid.powers (f r)) _ _).to_ring_equiv,
+  refine (hP.cancel_left_is_iso e₁.to_CommRing_iso.hom (CommRing.of_hom _)).symm.trans _,
+  refine (hP.cancel_right_is_iso (CommRing.of_hom _) e₂.to_CommRing_iso.hom).symm.trans _,
+  rw ← eq_iff_iff,
+  congr' 1,
+  dsimp [CommRing.of_hom, CommRing.of, bundled.of],
+  refine is_localization.ring_hom_ext (submonoid.powers r) _,
+  ext1,
+  revert e₁ e₂,
+  dsimp [ring_equiv.to_ring_hom, is_localization.away.map],
+  simp only [category_theory.comp_apply, ring_equiv.refl_apply, is_localization.alg_equiv_apply,
+    is_localization.ring_equiv_of_ring_equiv_apply, ring_hom.coe_mk, ring_equiv.to_fun_eq_coe,
+    is_localization.ring_equiv_of_ring_equiv_eq, is_localization.map_eq],
+end
+
+end respects_iso
+
+section stable_under_composition
+
+/-- A property is `stable_under_composition` if the composition of two such morphisms
+still falls in the class. -/
+def stable_under_composition : Prop :=
+  ∀ ⦃R S T⦄ [comm_ring R] [comm_ring S] [comm_ring T],
+    by exactI ∀ (f : R →+* S) (g : S →+* T) (hf : P f) (hg : P g), P (g.comp f)
+
+variable {P}
+
+lemma stable_under_composition.respects_iso (hP : ring_hom.stable_under_composition @P)
+  (hP' : ∀ {R S : Type*} [comm_ring R] [comm_ring S] (e : by exactI R ≃+* S),
+    by exactI P e.to_ring_hom) : ring_hom.respects_iso @P :=
+begin
+  split,
+  { introv H, resetI, apply hP, exacts [H, hP' e] },
+  { introv H, resetI, apply hP, exacts [hP' e, H] }
+end
+
+end stable_under_composition
+
+section stable_under_base_change
+
+/-- A morphism property is `stable_under_composition` if the composition of two such morphisms
+still falls in the class. -/
+def stable_under_base_change : Prop :=
```

Comment:
Copied docstring needs to be edited. This one isn't invariant under isomorphism, unlike the localization ones, because we don't have a characteristic predicate for tensor products ...

Reply:
Yeah but we would suppose that every property should at least `respects_iso` so I guess this is not that big of a deal?

Reply:
Sure!